src/lib/queries/vaults/useHistoricApr.ts

```diff
@@ -0,0 +1,52 @@
+import { useQuery } from "@tanstack/react-query";
+
+import { dayjs } from "@/lib/dayjs";
+import { QueryKeys } from "@/lib/queries/queriesSchema";
+import { ONE_DAY_IN_MS } from "@/lib/constants";
+import { useChain } from "@/hooks/useChain";
+import { ALCHEMIST_FEE_MULTIPLIER } from "@/lib/middleware/common";
+
+export const useHistoricApr = () => {
+  const chain = useChain();
+  return useQuery({
+    queryKey: [QueryKeys.HistoricYield, chain.id],
+    queryFn: async () => {
+      if (chain.id === 250)
+        throw new Error("Historic yield data is not supported on this chain");
+
+      const fileName =
+        chain.id === 1
+          ? "mainnetDailyAprs.csv"
+          : chain.id === 10
+            ? "optimismDailyAprs.csv"
+            : "arbitrumDailyAprs.csv";
+      const response = await fetch(
+        `https://api.pinata.cloud/data/pinList?includeCount=false&metadata[name]=${fileName}&status=pinned&pageLimit=1`,
+        {
+          method: "GET",
+          headers: {
+            Authorization: `Bearer ${import.meta.env.VITE_PINATA_KEY}`,
+          },
+        },
+      );
+      const data = await response.json();
+      const pinataHash = data.rows[0].ipfs_pin_hash as string;
```

Comment:
Super minor, but would change to data?.rows[0]. ipfs_pin_hash in case there's an error fetching the data

Reply:
I think this is not necessary since fetching error will throw at `fetch` and if response is not ok it will throw at `response.json`, so it won't even come to this

Reply:
I guess. I better put it this way: If we add optional chaining here, we will get an undefined `pinataHash` and it will throw two lines lower when we try to fetch, so I don't really see why we can't have a throw at this line.

Reply:
We should catch/handle any errors regardless of where they throw imo. If the structure of the data changes in the future then line 33 could throw an undefined reference error

Reply:
Sorry, I don't agree. It will just throw in this case and when we debug returned error from useQuery we will know right away that something is wrong with the data structure. 
There's no harm to add it tho, if you insist I can live with it.

Reply:
I suppose as long as a potential error there doesn't break the site it's not a huge deal

Reply:
I've just added an error message in UI in case of error. Is that what you meant?